Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/BPMNEditorController.java

```diff
@@ -435,7 +436,7 @@ private void pushViewModeParameters(final String bpmnXML) {
     param.put("viewOnly", true);
     param.put("langTag", getLanguageTag());
     param.put("doAutoLayout", "false");
-
+    param.put("username", currentUserType.getUsername());
```

Comment:
currentUserType is null at this point. We need to move the line where currentUserType is set (line 118) before the view link is opened (lines 112 - 116).